jar/src/clj/sixsq/slipstream/client/api/lib/app.clj

```diff
@@ -0,0 +1,90 @@
+(ns sixsq.slipstream.client.api.lib.app
+  ""
+  (:require
+    [superstring.core :as s]
+    [sixsq.slipstream.client.api.authn :refer [*context*]]
+    [sixsq.slipstream.client.api.impl.crud :as h]))
+
+;;
+(def param-refqname "refqname")
+(def param-scalable "mutable")
+(def param-keep-running "keep-running")
+(def param-tags "tags")
+(def param-type "type")
+
+(def params-reserved #{param-tags
+                       param-keep-running
+                       param-type})
+
+(defn assoc-comp-param
+  [m k v]
+  (let [k (s/trim (name k))]
+    (cond
+      (s/blank? k) m
+      (s/includes? k ":") (assoc m (apply format "parameter--node--%s--%s" (s/split k #":")) (str v))
+      :else (assoc m (format "parameter--%s" k) (str v)))))
```

Comment:
In this and the previous line it would be good to use `str` rather than `format` because `format` doesn't really exist in clojurescript.  For the simple formatting here, `str` shouldn't be too much of an inconvenience.